<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find maki_robot)/rosconsole.conf"/>
	<env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find maki_robot)/python_logging.conf"/>

	<arg name="behaviors" default="false" />
	<arg name="exp_controller" default="true" />
	<arg name="arbotix" default="false" />

  	<!-- robot interface -->
	<node name="maki_arbotix_interface" pkg="maki_robot" type="MAKI-Arbotix-Interface.py" output="screen" if="$(arg arbotix)" />

	<!-- controller -->
	<node name="inspire4_controller" pkg="maki_robot" type="INSPIRE4_controller.py" output="screen" if="$(arg exp_controller)" />

	<!-- behaviors -->
	<group if="$(arg behaviors)" >
		<node name="asleep_awake" pkg="maki_robot" type="asleep_awake.py" output="screen" />
		<node name="look_inspire4_intro" pkg="maki_robot" type="INSPIRE4_look_intro.py" output="screen" />
		<node name="look_inspire4_interactions" pkg="maki_robot" type="INSPIRE4_look_interactions.py" output="screen" />
		<node name="engagement_inspire4" pkg="maki_robot" type="INSPIRE4_engagement.py" output="screen" />
	</group>

	<node name="ros2vhmsg" pkg="ros2vhmsg"  type="ros2vhmsg" output="screen" args="--VHMSG_TOPIC 'maki_command;std_msgs/String' --VHMSG_TOPIC 'inspire_four_pilot_command;std_msgs/String'" />	
</launch>
